src/pages/index.tsx

```diff
@@ -2,128 +2,49 @@ import { FC, useState, useEffect, useContext } from 'react';
 import Head from '@/components/head';
 import Layout from '@/components/Layout';
 import useFetch from '@/hooks/useFetch';
+import Tabs from '@/components/Tabs/Tabs';
+import { TasksProvider } from '@/context/tasks.context';
+import getTaskMap from '@/helperFunctions/getTaskMap';
 import classNames from '@/styles/tasks.module.scss';
+import { TASKS_URL } from '@/components/constants/url';
 import task from '@/interfaces/task.type';
-import Accordion from '@/components/Accordion';
-import fetch from '@/helperFunctions/fetch';
-import { toast, ToastTypes } from '@/helperFunctions/toast';
-import {
-  ASSIGNED,
-  COMPLETED,
-  AVAILABLE,
-  IN_PROGRESS,
-  SMOKE_TESTING,
-  NEEDS_REVIEW,
-  IN_REVIEW,
-  APPROVED,
-  MERGED,
-  SANITY_CHECK,
-  REGRESSION_CHECK,
-  RELEASED,
-  VERIFIED,
-  BLOCKED,
-} from '@/components/constants/task-status';
-import updateTasksStatus from '@/helperFunctions/updateTasksStatus';
-import { useAppContext } from '@/context';
-import { isUserAuthorizedContext } from '@/context/isUserAuthorized';
-import { TasksProvider } from '@/context/tasks.context';
-import TaskList from '@/components/tasks/TaskList/TaskList';
-
-const { SUCCESS, ERROR } = ToastTypes;
-const TASKS_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/tasks`;
-const SELF_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/users/self`;
-const STATUS_ORDER = [
-  ASSIGNED,
-  COMPLETED,
-  BLOCKED,
-  AVAILABLE,
-  IN_PROGRESS,
-  SMOKE_TESTING,
-  NEEDS_REVIEW,
-  IN_REVIEW,
-  APPROVED,
-  MERGED,
-  SANITY_CHECK,
-  REGRESSION_CHECK,
-  RELEASED,
-  VERIFIED,
-];
-const statusActiveList = [
-  IN_PROGRESS,
-  BLOCKED,
-  SMOKE_TESTING,
-];
-async function updateCardContent(id: string, cardDetails: task) {
-  try {
-    const { requestPromise } = fetch({
-      url: `${TASKS_URL}/${id}`,
-      method: 'patch',
-      data: cardDetails,
-    });
-    await requestPromise;
-    toast(SUCCESS, 'Changes have been saved !');
-  } catch (err: any) {
-    if ('response' in err) {
-      toast(ERROR, err.response.data.message);
-      return;
-    }
-    toast(ERROR, err.message);
-  }
-}
-
 
 const Index: FC = () => {
-  const { state: appState } = useAppContext();  
-  const [filteredTask, setFilteredTask] = useState<any>([]);
+  const [filteredTasks, setFilteredTasks] = useState<null | task>(null);
```

Comment:
This state is not required, you can derive the state from the response itself.
After the error and isLoading condition simply extract and pass an array to tabs.[check below comment also]
```jsx
const tasks = response['tasks'] ?? [];
const taksMap = getTaskMap(tasks)
```